alicloud/resource_alicloud_cs_serveless_kubernetes.go

```diff
@@ -0,0 +1,309 @@
+	client := meta.(*connectivity.AliyunClient)
+	csService := CsService{client}
+	invoker := NewInvoker()
+	object, err := csService.DescribeCsServelessKubernetes(d.Id())
+	if err != nil {
+		if NotFoundError(err) {
+			d.SetId("")
+			return nil
+		}
+		return WrapError(err)
+	}
+
+	_ = d.Set("name", object.Name)
+	_ = d.Set("vpc_id", object.VpcId)
+	_ = d.Set("vswitch_id", object.VSwitchId)
+	_ = d.Set("security_group_id", object.SecurityGroupId)
+	_ = d.Set("private_zone", object.PrivateZone)
+	_ = d.Set("deletion_protection", object.DeletionProtection)
+	_ = d.Set("tags", object.Tags)
+
+	var requestInfo *cs.Client
+	var response interface{}
+
+	if err := invoker.Run(func() error {
+		raw, err := client.WithCsClient(func(csClient *cs.Client) (interface{}, error) {
+			requestInfo = csClient
+			return csClient.GetClusterCerts(d.Id())
+		})
+		response = raw
+		return err
+	}); err != nil {
+		return WrapErrorf(err, DefaultErrorMsg, d.Id(), "GetClusterCerts", DenverdinoAliyungo)
+	}
+	if debugOn() {
+		requestMap := make(map[string]interface{})
+		requestMap["ClusterId"] = d.Id()
+		addDebug("GetClusterCerts", response, requestInfo, requestMap)
+	}
+	cert, _ := response.(cs.ClusterCerts)
+	if ce, ok := d.GetOk("client_cert"); ok && ce.(string) != "" {
+		if err := writeToFile(ce.(string), cert.Cert); err != nil {
+			return WrapError(err)
+		}
+	}
+	if key, ok := d.GetOk("client_key"); ok && key.(string) != "" {
+		if err := writeToFile(key.(string), cert.Key); err != nil {
+			return WrapError(err)
+		}
+	}
+	if ca, ok := d.GetOk("cluster_ca_cert"); ok && ca.(string) != "" {
+		if err := writeToFile(ca.(string), cert.CA); err != nil {
+			return WrapError(err)
+		}
+	}
+
+	var config cs.ClusterConfig
+	if file, ok := d.GetOk("kube_config"); ok && file.(string) != "" {
+		var requestInfo *cs.Client
+
+		if err := invoker.Run(func() error {
+			raw, err := client.WithCsClient(func(csClient *cs.Client) (interface{}, error) {
+				requestInfo = csClient
+				return csClient.DescribeClusterUserConfig(d.Id(), d.Get("enndpoint_public_access_enabled").(bool))
+			})
+			response = raw
+			return err
+		}); err != nil {
+			return WrapErrorf(err, DefaultErrorMsg, d.Id(), "GetClusterConfig", DenverdinoAliyungo)
+		}
+		if debugOn() {
+			requestMap := make(map[string]interface{})
+			requestMap["ClusterId"] = d.Id()
+			addDebug("GetClusterConfig", response, requestInfo, requestMap)
+		}
+		config, _ = response.(cs.ClusterConfig)
+
+		if err := writeToFile(file.(string), config.Config); err != nil {
+			return WrapError(err)
+		}
+	}
+	return nil
+}
+
+func resourceAlicloudCSServelessKubernetesDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+	csService := CsService{client}
+	var requestInfo *cs.Client
+	invoker := NewInvoker()
+	var response interface{}
+
+	err := resource.Retry(30*time.Minute, func() *resource.RetryError {
```

Comment:
I think the `Retry` is redundant in the line 280 because of there is no any error needs to retry.

Reply:
removed